### PR TITLE
Feat/add test configuration

### PR DESCRIPTION
**Pull request by bearpersson**
## Explain the changes you’ve made
- Configured test settings in package.json
- Created folder for each component
- Added tests for button component
- Added tests for confirmation component

## Explain why these changes are made

To be able to add tests when implementing new featuers.

## How to test

Concrete example:

1. Checkout this branch
2. Swap the application to run storybook
3. Fire upp the simulator by running the command `yarn ios`
4. ...etc.

## Tested environments

- [] Your local environment.
- [] Staging environment.

## Screenshots

(Optional)

Screenshots are helpful for UI-related changes. It could be an before and after change as an example.
Even backend code can benefit from a screenshot of the net change.

---

## Checklist for reviewers

Here are a few points that are useful for reviewers to check.

The short list of the (maybe) most important points are:

- Does this PR work as expected?
- Does the PR properly only concern a single feature/bug/thing?
- Is the code clear and concise?
- Will the changes impact existing functionality?
- Are there any linting errors (ESLint) that should be resolved?
- Do the automatic GitHub checks pass?

However it is good if a reviewer can be as thorough as possible,
within reasonable limits of course.

### Functionality

- Does this PR work as expected?
- If this is a new feature, does it add value, or is it a sign of feature-creep?
- Will the changes impact existing functionality?
- Does the changes create inconsistencies somewhere else?
- Does the code work on different platforms if applicable?
- How does the code handle if unexpected values are passed to functions etc.?

### Readability, syntax, formatting, style

- Is the code clear and concise?
- Has styling rules (Prettier) been applied to included files?
- Are functions/variables/etc. aptly named?

### Design

- Is the code properly planned and designed?
- Is code organized/separated in a logical fashion?
- Does the code follow clean code practices to a reasonable degree?
- Is the code self-documenting enough?
- Is the code in sync with existing patterns/technologies?
- Does the code reuse existing functionality where applicable?
- Is the code designed with reusability in mind?
- Does the code have hard-coded values that can be refactored?

### Patterns, idioms, tests, best practices

- Are there any linting errors (ESLint) that should be resolved?
- Is the code properly typed (TypeScript)?
- Does the code utilize language features to their full potential?

### Debugging, testing, configuring

- Is appropriate logging used?
- Have tests been added/modified to accommodate the changes?
- Is the feature sufficiently configurable in case tweaks are required in the future?

### Performance, reliability, scalability

- Is the code well optimized?
- Does the code scale?
- Does the code handle failure scenarios?

### Security

- Is there any obvious potential for exploits and security threats?
- Does code check proper authorization where applicable?

### Etiquettes

- Are commits atomic (does only one thing per commit)?
- Does the PR properly only concern a single feature/bug/thing?
- Are commit messages well-written?
- Do the automatic GitHub checks pass?

### CODE REVIEW

File: source/js/components/Button/Button.test.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { fireEvent, render } from '@testing-library/react';
+import { Button } from '.';
+
+test('Should render button with label text', () => {
+  const element = render(<Button id="button" label="MyLabel" type="button" />).container;
+  expect(element).toHaveTextContent('MyLabel');
+});
+
+test.each`
```
Comment (karlknowit) [inline source/js/components/Button/Button.test.tsx:9]: https://jestjs.io/docs/api#1-testeachtablename-fn-timeout

File: source/js/components/Button/Button.test.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { fireEvent, render } from '@testing-library/react';
+import { Button } from '.';
+
+test('Should render button with label text', () => {
```
Comment (karlknowit) [inline source/js/components/Button/Button.test.tsx:4]: Arrange, Act, Assert syntax

**Review by karlknowit — CHANGES_REQUESTED** [review]

File: source/js/components/Confirmation/Confirmation.test.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+  expect(element).toHaveTextContent('Tack för din bokning!');
+});
+
+it('informs user on who to meet', () => {
+  const screen = render(<Confirmation {...mockedProps} />);
+
+  const element = screen.getByText(/Du har bokat tid med Firstname Lastname följande tid:/i);
+
+  expect(element).toBeInTheDocument();
```
Comment (karlknowit) [inline source/js/components/Confirmation/Confirmation.test.tsx:25]: toBeVisible?

File: source/js/components/Confirmation/Confirmation.test.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+it('renders a title', () => {
+  const container = render(<Confirmation {...mockedProps} />).container;
+
+  const element = container.querySelector('p.c-typography__variant--h1');
+
+  expect(element).toHaveTextContent('Tack för din bokning!');
+});
+
+it('informs user on who to meet', () => {
+  const screen = render(<Confirmation {...mockedProps} />);
```
Comment (karlknowit) [inline source/js/components/Confirmation/Confirmation.test.tsx:21]: const { getByText } = render ...

or import screen:
```js
import { render, screen } from '@testing-library/react';

const { getByText } = screen;

it (.., () => {
 render ...
getByText(...)
})
 

```

File: source/js/components/Confirmation/Confirmation.tsx [hunk 5]
```diff
@@ -1,21 +1,19 @@
...
   userEmail: string;
-  date: string;
-  startTime: string;
-  endTime: string;
+  date?: string;
+  startTime?: string;
+  endTime?: string;
 }
-const Confirmation = ({ administratorName, userEmail, date, startTime, endTime }: ConfirmationInterface) => (
+export const Confirmation = ({ administratorName, userEmail, date, startTime, endTime }: ConfirmationInterface) => (
```
Comment (karlknowit) [inline source/js/components/Confirmation/Confirmation.tsx:10]: consider default the import params instead:

```js
export const Confirmation = ({ administratorName, userEmail, date = '', startTime = '', endTime = '' }: ConfirmationInterface)
```

by doing that we don't need to have logic like:
```js
date ?? ''
```
which means that our function calls gets a little bit cleaner:

```js
formatTimePeriod(date , startTime, endTime)
```

File: source/js/components/Form/Form.tsx [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import { FormEventHandler } from 'react';
```
Comment (karlknowit) [inline source/js/components/Form/Form.tsx:1]: Missing tests for for, but that's perhaps fine to begin with

**Review by karlknowit — CHANGES_REQUESTED** [review]

File: source/js/components/Form/Form.tsx (hunk 6 above)
Comment (bearpersson) [reply source/js/components/Form/Form.tsx:1]: Yes me too. We have to add more tests over time, it felt out of scope to add tests for every component for this task.

File: source/js/components/Confirmation/Confirmation.test.tsx (hunk 4 above)
Comment (bearpersson) [reply source/js/components/Confirmation/Confirmation.test.tsx:21]: Fixed

File: source/js/components/Confirmation/Confirmation.tsx (hunk 5 above)
Comment (bearpersson) [reply source/js/components/Confirmation/Confirmation.tsx:10]: I agree. Fixed

**Review by karlknowit — APPROVED** [review]

**Pull request merged**